tests/analysis/multi_delimited_token_stream_tests.cpp

```diff
@@ -0,0 +1,242 @@
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include "analysis/multi_delimited_token_stream.hpp"
+#include "gtest/gtest.h"
+#include "tests_config.hpp"
+
+namespace {
+
+irs::bstring operator""_b(const char* ptr, std::size_t size) {
+  return irs::bstring{
+    irs::ViewCast<irs::byte_type>(std::string_view{ptr, size})};
+}
+
+class multi_delimited_token_stream_tests : public ::testing::Test {
+  virtual void SetUp() {
+    // Code here will be called immediately after the constructor (right before
+    // each test).
+  }
+
+  virtual void TearDown() {
+    // Code here will be called immediately after each test (right before the
+    // destructor).
+  }
+};
+
+}  // namespace
+
+// -----------------------------------------------------------------------------
+// --SECTION--                                                        test suite
+// -----------------------------------------------------------------------------
+
+TEST_F(multi_delimited_token_stream_tests, consts) {
+  static_assert("multi_delimiter" ==
+                irs::type<irs::analysis::MultiDelimitedAnalyser>::name());
+}
+
+TEST_F(multi_delimited_token_stream_tests, test_delimiter) {
+  auto stream =
+    irs::analysis::MultiDelimitedAnalyser::make({.delimiters = {"a"_b}});
+  ASSERT_EQ(irs::type<irs::analysis::MultiDelimitedAnalyser>::id(),
+            stream->type());
+
+  ASSERT_TRUE(stream->reset("baccaad"));
+
+  auto* payload = irs::get<irs::payload>(*stream);
+  ASSERT_EQ(nullptr, payload);
+  auto* term = irs::get<irs::term_attribute>(*stream);
+  auto* inc = irs::get<irs::increment>(*stream);
+  auto* offset = irs::get<irs::offset>(*stream);
+  ASSERT_EQ(inc->value, 1);
+
+  ASSERT_TRUE(stream->next());
+  ASSERT_EQ("b", irs::ViewCast<char>(term->value));
+  ASSERT_EQ(offset->start, 0);
+  ASSERT_EQ(offset->end, 1);
+  ASSERT_TRUE(stream->next());
+  ASSERT_EQ("cc", irs::ViewCast<char>(term->value));
+  ASSERT_EQ(offset->start, 2);
+  ASSERT_EQ(offset->end, 4);
+  ASSERT_TRUE(stream->next());
+  ASSERT_EQ("d", irs::ViewCast<char>(term->value));
+  ASSERT_EQ(offset->start, 6);
+  ASSERT_EQ(offset->end, 7);
+  ASSERT_FALSE(stream->next());
+}
+
+TEST_F(multi_delimited_token_stream_tests, test_delimiter_empty_match) {
+  auto stream =
+    irs::analysis::MultiDelimitedAnalyser::make({.delimiters = {"."_b}});
+  ASSERT_EQ(irs::type<irs::analysis::MultiDelimitedAnalyser>::id(),
+            stream->type());
+
+  ASSERT_TRUE(stream->reset(".."));
+
+  auto* payload = irs::get<irs::payload>(*stream);
+  ASSERT_EQ(nullptr, payload);
+
+  ASSERT_FALSE(stream->next());
+}
+
+TEST_F(multi_delimited_token_stream_tests, test_delimiter_5) {
```

Comment:
We need additional case with 3 "single" byte delimiters. As we have a special case - see MultiDelimitedTokenStreamSingleChars<N> instantiation for cases N > 1  & N < 4. And this is not covered.